### ISSUE TITLE
Spiral vase works fine on vertical objects, but gives compensation artefacts on slopes

### ISSUE THREAD

**Issue by Leozolt** [alexrj/Slic3r#4360]
The error looks like the slicer makes a spiral on the Z layer (
![helicoidal](https://user-images.githubusercontent.com/26518970/38216055-e5d150c2-36c9-11e8-8527-1c4f107d584e.png)
) 

but compensates the **diameter** of the spiral on a single spot, instead of doing a conical spiral (
![conicalspiral](https://user-images.githubusercontent.com/26518970/38216067-f3556ecc-36c9-11e8-85e0-f535e0ca4f6e.jpg)
).

Using slic3r 1.3.0-dev but had the same issue before tho it's less visible with a thinner nozzle (0,4 .. 0,6mm).

Using 1,2mm nozzle in the print pictures also happens if there is *no seam* in the stl on the spot where it happens.

The error appears at the same spot where the spiral begins.

![spiralvaseoverhang](https://user-images.githubusercontent.com/26518970/38216300-abdf0d18-36ca-11e8-9d44-cb5cc60bfafa.jpg)
![dsc01248_li](https://user-images.githubusercontent.com/26518970/38216384-e7818224-36ca-11e8-95e1-d4c2d894d0df.jpg)

It's not visible in preview or layer preview as they only show layer circles an no actual path preview...
![preview](https://user-images.githubusercontent.com/26518970/38217387-79c9f53c-36ce-11e8-85bc-b86849337ddb.jpg)

If a stl is required i prefer transmitting it on a non public space for this model.

Thank You!

**Comment by lordofhyphens** [comment]
@Leozolt could you add a copy of the configuration? Use CTRL-E (or File->Export Config)

**Comment by lordofhyphens** [comment]
Here's a possible test case @Leozolt can you confirm this model with your config causes the problem?

[possible_test.zip](https://github.com/slic3r/Slic3r/files/1869665/possible_test.zip)

**Comment by Leozolt** [comment]
Printed out the slopevase.stl
![dsc01257](https://user-images.githubusercontent.com/26518970/38222101-3e60d69a-36e3-11e8-8378-d84d6b7502eb.JPG)

The change in diameter per "layer" is clearly visible/feelable on the issue line but i am not able to photograph it accordingly.

Same issue with my config (needed to reduce to 15mm/s for the slopevase the diameter is too small) 
[config.zip](https://github.com/slic3r/Slic3r/files/1869773/config.zip)

**Comment by Leozolt** [comment]
Is it possible to apply the "hack" on the x y plane to make the diameter grow continually?

**Comment by bubnikv** [comment]
Maybe. One would have to find corresponding points between two successive contours and then to linearly  morph from one contour to the other. This is not too difficult to implement.

Other solution would be to calculate the spiral over the input surface, but that is much more complex to do.

**Comment by Leozolt** [comment]
ok, it would require to calculate a spiral (conical helix segment) from one layer to the next and subdivide it to small linear segments. ??

**Comment by mesheldrake** [comment]
No, not always a conical helix segment. Think of the general case - any old spiral vase type model - where it's not necessarily conical. That's where @bubnikv 's first suggestion makes more sense.
